src/fieldsArray.js

```diff
@@ -6,98 +6,65 @@ const debug = require("debug")("passkit:fields");
  * @see https://apple.co/2wkUBdh
  */
 
-class FieldsArray extends Set {
-  constructor(fields) {
-    super();
-
-    /**
-     * Pass fields must be unique (for key) in its scope.
-     * Therefore we use a Set to keep them tracked.
-     */
-
-    this.fieldsKeys = new Set();
-    fields.forEach(a => this[a] = new ItemsArray(this));
-  }
-
-  addFieldKey(key) {
-
-    if (this.fieldsKeys.has(key)) {
-      debug(`Field with key "${key}" discarded: fields must be unique in pass scope.`);
-    } else {
-      this.fieldsKeys.add(key);
-    }
-  }
-
-  deleteFieldKey(key) {
-    this.fieldsKeys.delete(key);
-  }
-
-  emptyUnique() {
-    this.fieldsKeys.clear();
-  }
+const poolSymbol = Symbol("pool");
+
+class FieldsArray extends Array {
+	
+	constructor(pool,...args) {
+    	super(...args);
```

Comment:
@ianbale please look here, at the rows with wrong indentation: they are still using spaces. Just to let you know 😉